app/tests/spec/views/sign_in.js

```diff
@@ -593,7 +598,81 @@ function (chai, $, sinon, p, View, Session, AuthErrors, Metrics, FxaClient,
         return view.render()
           .then(function () {
             assert.ok($('.avatar-view').length, 'should show suggested avatar');
-            assert.notOk($('.password').length, 'should show password input');
+            assert.notOk($('.password').length, 'should not show password input');
+            assert.isTrue(TestHelpers.isEventLogged(metrics, 'signin.ask-password.skipped'));
+          });
+      });
+
+      it('asks for the password if the stored session is not from sync', function () {
+        var account = user.initAccount({
+          sessionToken: 'abc123',
+          email: 'a@a.com',
+          verified: true,
+          accessToken: 'foo'
+        });
+
+        sinon.stub(view, 'getAccount', function () {
+          return account;
+        });
+
+        relier.set('service', 'loop');
+
+        return view.render()
+          .then(function () {
+            assert.equal($('.email')[0].type, 'hidden', 'should not show email input');
```

Comment:
Instead of `$('.email')[0].type`, you can make this a bit simpler: `$('.email').attr('type')`